### PR TITLE
Add AbstractRadioUnit and RadioUnitState

### PR DESCRIPTION
**Pull request by dLeung09**
* Added the AbstractRadioUnit abstract class, which is used by the
concrete implementors of the RadioUnitState interface.

* Added the concrete implementors of the RadioUnitState interface
(IdleState, DeactivatedState, ActivatedState) and implements the logic
for the state pattern

* Added the Vendor enum

* Added the IllegalStateTransitionException for handling improper state
transitions

### CODE REVIEW

File: src/main/java/common/Vendor.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/**
```
Comment (matthewkwsiu) [inline src/main/java/common/Vendor.java]: Minor: These comment lines could be removed.

**Review by john-breton — APPROVED** [review]

File: src/main/java/radiounit/AbstractRadioUnit.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+/**
```
Comment (matthewkwsiu) [inline src/main/java/radiounit/AbstractRadioUnit.java]: Minor: remove extra comment lines

File: src/main/java/radiounit/radiostate/IllegalStateTransitionException.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/**
```
Comment (matthewkwsiu) [inline src/main/java/radiounit/radiostate/IllegalStateTransitionException.java]: Minor: remove extra comment lines

File: src/main/java/radiounit/radiostate/IllegalStateTransitionException.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+package radiounit.radiostate;
+
+/**
+ * @author edavleu
+ *
+ */
+public class IllegalStateTransitionException extends Exception {
+
+	/**
+	 * @param format
```
Comment (matthewkwsiu) [inline src/main/java/radiounit/radiostate/IllegalStateTransitionException.java]: What did you mean by format? As in, the "message" string being passed in is responsible for creating the format? My understanding was that the "message" string is a message saved for later retrieval, but I'm not very familiar with exceptions.

File: src/main/java/radiounit/radiostate/IdleState.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
+/**
```
Comment (matthewkwsiu) [inline src/main/java/radiounit/radiostate/IdleState.java]: Minor: extra comment lines

File: src/main/java/radiounit/radiostate/DeactivatedState.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
+/**
```
Comment (matthewkwsiu) [inline src/main/java/radiounit/radiostate/DeactivatedState.java]: Minor: Extra comment lines

File: src/main/java/radiounit/radiostate/ActivatedState.java [hunk 7]
```diff
@@ -0,0 +1,102 @@
+/**
```
Comment (matthewkwsiu) [inline src/main/java/radiounit/radiostate/ActivatedState.java]: Minor: Extra comment lines

**Review by matthewkwsiu — APPROVED** [review]
Looks good to me. Just some minor comments about comment lines (there were a few empty comment lines at the beginning of most files) and a question about the name of a parameter.

File: src/main/java/radiounit/radiostate/IllegalStateTransitionException.java (hunk 4 above)
Comment (dLeung09) [reply src/main/java/radiounit/radiostate/IllegalStateTransitionException.java]: This was auto-generated when it created the constructor - I'll change

**Comment by matthewkwsiu** [comment]
Great, looks good to me!

**Pull request merged**